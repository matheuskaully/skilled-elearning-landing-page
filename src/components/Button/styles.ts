import styled, { css } from 'styled-components';

interface ButtonProps {
  backgroundColor: string;
  hoverColor: string;
}

export const Button = styled.button<ButtonProps>` 
  background: ${(props) => props.backgroundColor};
  
  &:hover {
    background: ${(props) => props.hoverColor};
  }

  ${({ theme }) => css`
    width: 167px;
    height: 56px;
    border-radius: 28px;
    border: none;
    font-size: 18px;
    font-weight: 700;
    line-height: 28px;
    letter-spacing: 0px;
    color: ${theme.colors.white};
    
    cursor: pointer;

    &:hover {
      opacity: 80%;
      transition: ${theme.transition.fast};
      background: ${theme.colors.lightPurple};
    }
  `}
`;
