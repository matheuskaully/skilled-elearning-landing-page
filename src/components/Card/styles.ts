import styled, { css } from "styled-components";

export const Container = styled.div`
  ${({ theme }) => css`
    width: 350px;
    height: 322px;
    padding: 64px 30px 40px 30px;
    margin: 0 15px;
    border-radius: 15px;
    display: flex;
    flex-direction: column;

    img {
      position: relative;
      width: 56px;
      height: 56px;
      bottom: 280px;
    }

    span {
      font-size: 24px;
      font-weight: 800;
      line-height: 30px;
      letter-spacing: 0px;
      font-weight: 800;
      font-size: 24px;
      line-height: 30px;
      color: ${theme.colors.secondary};
    }

    p {
      padding-top: 24px;
      font-weight: 500;
      font-size: 18px;
      line-height: 28px;
      color: ${theme.colors.gray};
    }

    a {
      position: relative;
      bottom: -40px;
      text-decoration: none;
      font-weight: 700;
      font-size: 18px;
      line-height: 28px;
      color: ${theme.colors.primary};

      &:hover {
        opacity: 60%;
        transition: ${theme.transition.fast};
      }
    }
  `}
`