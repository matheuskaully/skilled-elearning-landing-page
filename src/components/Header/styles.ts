import styled, { css } from 'styled-components';
import media from 'styled-media-query';

export const Container = styled.header`
  ${({ theme }) => css`
    max-width: ${theme.grid.container};
    width: 100%;
    margin-top: ${theme.spacings.medium};
    display: flex;
    align-items: center;
    justify-content: space-between;
    margin-inline: auto;
    padding-inline: ${theme.spacings.xsmall};

    ${media.lessThan('medium')`
      margin-top: ${theme.spacings.small};
    `}

    ${media.lessThan('small')`
      margin-top: ${theme.spacings.xsmall};
    `}
  `}
`;

export const Cards = styled.div`
  display: flex;
`;
