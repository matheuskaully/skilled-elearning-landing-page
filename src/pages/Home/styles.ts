import styled, { css } from 'styled-components';

export const Container = styled.div`
  position: relative;
  height: 100vh;
  overflow-x: hidden;
`;

export const Image = styled.img`
  position: absolute;
  top: -123px;
  right: -346px;
`;

export const Grid = styled.div`
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  gap: 40px;
`;
