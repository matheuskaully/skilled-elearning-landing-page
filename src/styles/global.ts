import { createGlobalStyle } from "styled-components"

export const GlobalStyle = createGlobalStyle`
  :root {
    --purple: #666ca3;
    --black: #13183F;
    --gray: #83869a;
    --pink1: #f74780;
    --pink2: #ffa7c3;
    --white: #fff;
    --gradient1: linear-gradient(180deg, #ff6f48, #f02aa6);
    --gradient2: linear-gradient(180deg, #f02aa6, #4851ff);
  }

  * {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
    font-family: 'Plus Jakarta Sans', sans-serif;
  }
`