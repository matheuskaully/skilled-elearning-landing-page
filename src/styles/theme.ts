import { DefaultTheme } from 'styled-components';

export const theme: DefaultTheme = {
  grid: {
    container: '110rem',
    gutter: '2.4rem'
  },
  border: {
    radius: {
      small: '1.5rem',
      medium: '2.8rem'
    }
  },
  font: {
    family:
      "Plus Jakarta Sans, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif",
    medium: 500,
    bold: 700,
    extraBold: 800,
    sizes: {
      small: '1.6rem',
      medium: '1.8rem',
      large: '2.4rem',
      xlarge: '3.2rem',
      xxlarge: '4.0rem',
      huge: '5.6rem'
    }
  },
  lineHeights: {
    medium: '2.8rem',
    large: '2.8rem',
    xlarge: '4.0rem',
    xxlarge: '5.1rem',
    huge: '7.0rem'
  },
  colors: {
    primary: '#F74780',
    secondary: '#13183F',
    lightPurple: '#666CA3',
    gray: '#83869A',
    white: '#FFFFFF',
    pink: '#FFA7C3',
    primaryGradient: '180deg, #FF6F48 0.04%, #F02AA6 100%',
    secundaryGradient: '180deg, #4851FF 0%, #F02AA6 99.92%'
  },
  spacings: {
    xxsmall: '0.8rem',
    xsmall: '1.6rem',
    small: '2.4rem',
    medium: '3.2rem',
    large: '4.0rem',
    xlarge: '4.8rem',
    xxlarge: '5.6rem'
  },
  layers: {
    base: 10,
    menu: 20,
    overlay: 30,
    modal: 40,
    alwaysOnTop: 50
  },
  transition: {
    default: '0.3s ease-in-out',
    fast: '0.1s ease-in-out'
  }
};
