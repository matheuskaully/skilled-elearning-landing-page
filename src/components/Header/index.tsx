import * as S from './styles';
import { Button } from '../Button';
import logoImg from '../../assets/logo-dark.svg';
import { theme } from '../../styles/theme';

export default function Header() {
  return (
    <S.Container>
      <img src={logoImg} alt="skilled" />

      <Button
        backgroundColor={theme.colors.secondary}
        hoverColor={theme.colors.lightPurple}
      >
        Get Started
      </Button>
    </S.Container>
  );
}
