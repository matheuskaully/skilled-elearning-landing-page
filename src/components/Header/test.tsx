import { screen } from '@testing-library/react';

import Header from '.';

import { renderWithTheme } from '../../styles/renderTheme';

describe('Header', () => {
  it('should render header component', () => {
    const { container } = renderWithTheme(<Header />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render image', () => {
    renderWithTheme(<Header />);

    const image = screen.getByRole('img', { name: 'skilled' });

    expect(image).toHaveAttribute('src', '/src/assets/logo-dark.svg');
  });

  it('should render button', () => {
    renderWithTheme(<Header />);

    const button = screen.getByRole('button', { name: 'Get Started' });

    expect(button).toBeInTheDocument();
  });
});
