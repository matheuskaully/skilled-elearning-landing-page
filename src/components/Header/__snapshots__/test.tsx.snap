// Vitest Snapshot v1

exports[`Header > should render header component 1`] = `
.c1 {
  background: linear-gradient(180deg,#FF6F48 0.04%,#F02AA6 100%);
  line-height: 2.4rem;
  border-radius: 28px;
  padding: 1.6rem 3.2rem;
  border: none;
  font-size: 18px;
  font-weight: 700;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  color: #FFFFFF;
  cursor: pointer;
  -webkit-transition: background 0.3s ease-in-out;
  transition: background 0.3s ease-in-out;
}

.c1:hover {
  opacity: 80%;
}

.c0 {
  max-width: 110rem;
  width: 100%;
  margin-top: 3.2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-inline: auto;
  padding-inline: 1.6rem;
}

@media (max-width:768px) {
  .c0 {
    margin-top: 2.4rem;
  }
}

@media (max-width:450px) {
  .c0 {
    margin-top: 1.6rem;
  }
}

<header
  class="c0"
>
  <img
    alt="skilled"
    src="/src/assets/logo-dark.svg"
  />
  <button
    class="c1"
  >
    Get Started
  </button>
</header>
`;
